pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,309 @@
+                - "type": condition that will trigger the acquisition of waveforms [EDGE,
+                slew,GLIT,intv,runt,drop]
+                - "source": trigger source (str, {EX,EX/5,C1,C2}) 
+                - "hold_type": hold type (refer to page 131 of programing guide)
+                - "hold_value1": hold value1 (refer to page 131 of programing guide)
+                - "level": Level at which the trigger will be set (float)
+                - "slope": (str,{POS,NEG,WINDOW}) Triggers on rising, falling or Window.
+                - "mode": behavior of the trigger following a triggering event (str, {NORM, AUTO, SINGLE,STOP}) 
+                - "coupling":  (str,{AC,DC}) Coupling to the trigger channel
+        Returns a flag indicating if all specified entries were correctly set on the oscilloscope and updates the interal trigger configuration
+        """
+        triggerConfDict=self.getTriggerConfig()
+        #self.triggerConfDict=triggerConfDict
+        setProcesses={
+            "setup":lambda dictIn: (
+                dictIn.get("type"),
+                dictIn.get("hold_type"),
+                dictIn.get("hold_value1")
+            ),
+            "level":lambda dictIn: "%.2eV"%dictIn.get("level"),
+            "coupling": lambda dictIn: dictIn.get("coupling"),
+            "slope": lambda dictIn: dictIn.get("slope"),
+            "mode": lambda dictIn: dictIn.get("mode")
+        }
+        setCommands={
+            "setup":"TRSE %s,SR,{ch},HT,%s,HV,%s",
+            "level":"{ch}:TRLV %s",
+            "coupling":"{ch}:TRCP %s",
+            "slope":"{ch}:TRSL %s",
+            "mode":"TRMD %s"
+        }
+        setValues={# For a given change in conf dict,find the relevant command to be called
+            "setup":["source","type","hold_type","hold_value1"],
+            "level":["level"],
+            "coupling":["coupling"],
+            "slope":["slope"],
+            "mode":["mode"]
+        }
+        if kwargs.get('source') is not None:
+            self.id=kwargs['source']
+        changedValues=[key for key in kwargs if not triggerConfDict[key]==kwargs[key]]
+        processToChange=[key for key in setValues if any([value in changedValues for value in setValues[key]])]
+        for changedKey in changedValues:
+            triggerConfDict[changedKey]=kwargs[changedKey]
+        for processKey in processToChange:
+            self.write(setCommands[processKey]%setProcesses[processKey](triggerConfDict))
+        self.triggerConfDict=self.getTriggerConfig()
+        statusFlag=self.triggerConfDict==triggerConfDict
+        return statusFlag
+
+
+class SDS1072CML(Instrument):
+    """ Represents the SIGLENT SDS1072CML Oscilloscope
+    and provides a high-level for interacting with the instrument
+    """
+    def __init__(self, adapter, name="Siglent SDS1072CML Oscilloscope", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            **kwargs
```

Comment:
Please state, whether the device supports SCPI (in this case inherit `SCPIMixin` before `Instrument`) or not (in this case set `includeSCPI=False`).